Sources/SemanticAnalyzer/SemanticAnalyzer.swift

```diff
@@ -355,21 +355,33 @@ public struct SemanticAnalyzer: ASTPass {
   }
 
   public func process(binaryExpression: BinaryExpression, passContext: ASTPassContext) -> ASTPassResult<BinaryExpression> {
+    let environment = passContext.environment!
     var binaryExpression = binaryExpression
 
     if case .dot = binaryExpression.opToken {
       // The identifier explicitly refers to a state property, such as in `self.foo`.
       // We set its enclosing type to the type it is declared in.
       let enclosingType = passContext.enclosingTypeIdentifier!
-      let lhsType = passContext.environment!.type(of: binaryExpression.lhs, enclosingType: enclosingType.name, scopeContext: passContext.scopeContext!)
+      let lhsType = environment.type(of: binaryExpression.lhs, enclosingType: enclosingType.name, scopeContext: passContext.scopeContext!)
       binaryExpression.rhs = binaryExpression.rhs.assigningEnclosingType(type: lhsType.name)
     }
 
     return ASTPassResult(element: binaryExpression, diagnostics: [], passContext: passContext)
   }
 
   public func process(functionCall: FunctionCall, passContext: ASTPassContext) -> ASTPassResult<FunctionCall> {
-    return ASTPassResult(element: functionCall, diagnostics: [], passContext: passContext)
+    let environment = passContext.environment!
+    var diagnostics = [Diagnostic]()
+
+    if environment.isInitializerCall(functionCall),
+      !passContext.inAssignment,
+      !passContext.isPropertyDefaultAssignment,
+      functionCall.arguments.isEmpty
+    {
+      diagnostics.append(.noReceiverForStructInitializer(functionCall))
+    }
```

Comment:
How does this relate?

Reply:
It doesn't really relate to the operand types, but it's an issue I came across while working on this PR. It's in its own commit so won't mess up the history.

Reply:
I thought it was small enough not to be a worry but I will do so.